main.c

```diff
@@ -9,128 +10,413 @@
-    BookCopy* copy5 = createBookCopy(05, "567", active);
-    BookCopy* copy6 = createBookCopy(06, "8910", active);
-    insertBookCopy(copy1, &inventory);
-    insertBookCopy(copy2, &inventory);
-    insertBookCopy(copy3, &inventory);
-    insertBookCopy(copy4, &inventory);
-    insertBookCopy(copy5, &inventory);
-    insertBookCopy(copy6, &inventory);
-    printf("***INVENTORY***\n");
-    printBookCopies(inventory);
-
-    // Rent book cases:
-    rentBook(new_book1, new_user1);
-    rentBook(new_book2, new_user1);
-    rentBook(new_book2, new_user2);
-    rentBook(new_book2, new_user3);
-
-    printf("***TRANSACTIONS***\n");
-    printTransactions(transaction_list);
-
-    printf("***Updated Library and Inventory after Renting***\n");
-    printBooks(library);
-    printBookCopies(inventory);
-
-    returnBook(04);
-    printf("***Updated Library and Inventory after Returning***\n");
-    printBooks(library);
-    printBookCopies(inventory);
-
-    printf("***TRANSACTIONS***\n");
-    printTransactions(transaction_list);
-
-    BookTransaction* searched_trans = searchTransaction("Luke", transaction_list, user_list_head); // use name to find user id linked w trans
-    printf("*****Searched Transaction Results*****\n");
-    printTransactions(searched_trans);
-    BookTransaction* trans_parameters = createTransaction(0, getTodaysDate(), 0, no_trans_status);
-    BookTransaction* filtered_trans = filterTransactions(trans_parameters, transaction_list);
-    printf("*****Filtered Transaction Results*****\n");
-    printTransactions(filtered_trans);
-
-    freeTransactionList(searched_trans);
-    freeTransactionList(filtered_trans);
-
-    char* login_name = getLoginName();
-    char* password = getPassword();
-    login(login_name, password);
-    // Do something in between
-    free(login_name);
-    free(password);
-    logout();*/
-
     // FILE MANAGEMENT
     readRecord(book_file, &book_record_head);
-    printRecord(book_record_head);
     loadBooks(book_record_head, &library);
-    printBooks(library);
-    saveBooks(book_file, library);
 
     readRecord(book_copy_file, &book_copy_record_head);
-    printRecord(book_copy_record_head);
     loadBookCopies(book_copy_record_head, &inventory);
-    printBookCopies(inventory);
-    saveBookCopies(book_copy_file, inventory);
 
     readRecord(user_file, &user_record_head);
-    printRecord(user_record_head);
     loadUsers(user_record_head, &user_list_head);
-    printUser(user_list_head);
-    saveUsers(user_file, user_list_head);
 
     readRecord(transaction_file, &transaction_record_head);
-    printRecord(transaction_record_head);
     loadTransactions(transaction_record_head, &transaction_list);
-    printTransactions(transaction_list);
-    saveTransactions(transaction_file, transaction_list);
 
+    printf("Welcome to the Library Management System!\n");
+    // AUTHENTICATION
+    char* log_in_name;
+    char* password;
+    int attempt = 0;
```

Comment:
Although it will work in our use case, but practically we need to implement it other way